### ISSUE TITLE
"retryLimit appears unused" but its used.

### ISSUE THREAD

**Issue by jkroepke** [koalaman/shellcheck#2186]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

retryLimit=3

while ! false; do
  echo "Failed! Retry in 20s ..."

  if [[ "((--retryLimit))" -le 0 ]]; then
    break;
  fi
  sleep 2
done

```

#### Here's what shellcheck currently says:
```
Line 3	SC2034: retryLimit appears unused. Verify use (or export if used externally).
```
#### Here's what I wanted or expected to see:

No issue.

**Comment by koalaman** [comment]
You're right, this was too dynamic for ShellCheck to figure out. The operands to `-le` are interpreted as arithmetic expressions, but ShellCheck currently does not re-parse the string argument as an expression. 

You can consider doing e.g. `if (( --retrylimit ))` (or more explicitly `(( --retrylimit <= 0 ))`) instead as a workaround.
